mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl

```diff
@@ -1462,10 +1461,36 @@ func getAllStatusBut(status string) []string {
 	return computeInstanceStatus
 }
 
-func waitUntilInstanceHasDesiredStatus(config *transport_tpg.Config, d *schema.ResourceData) error {
-	desiredStatus := d.Get("desired_status").(string)
+func changeInstanceStatusOnCreation(config *transport_tpg.Config, d *schema.ResourceData, project, zone, status, userAgent string) error {
```

Comment:
apologies for the late reply. Here's the log output:
```hcl
=== RUN   TestAccDataSourceComputeInstanceSerialPort_basic
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc001cae180] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc0013ea300] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc0013e3e00] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc0004ffc80] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc001cae000] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc001cae300] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc001cae000] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc001cae300] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc0004ffc80] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc0013ead80] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc0013e3e00] Closing: EOF 
2024/10/03 17:36:47 [DEBUG] [transport] [server-transport 0xc001caf500] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc0013eaf00] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc001caf380] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc001caf680] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc001caf680] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc001caf380] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc0013eaf00] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc000002480] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc000002780] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc000002780] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc000002480] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc000003080] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc000003080] loopyWriter exiting with error: transport closed by client 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc001cae000] Closing: EOF 
2024/10/03 17:36:48 [DEBUG] [transport] [server-transport 0xc001cae000] Closing: EOF 
panic: reflect: call of reflect.Value.NumField on zero Value

goroutine 3230 [running]:
reflect.flag.mustBe(...)
	/usr/local/go/src/reflect/value.go:233
reflect.Value.NumField({0x0?, 0x0?, 0x437c820?})
	/usr/local/go/src/reflect/value.go:2062 +0x85
github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource.setOmittedFields({0x437c820?, 0x0?}, {0x437c820?, 0xc0022fe1c0})
	/go/src/github.com/modular-magician/terraform-provider-google-beta/google-beta/tpgresource/convert.go:60 +0x173
github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource.Convert({0x437c820?, 0x0}, {0x437c820, 0xc0022fe1c0})
	/go/src/github.com/modular-magician/terraform-provider-google-beta/google-beta/tpgresource/convert.go:27 +0x6f
github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute.ComputeOperationWaitTime(0xc002976400, {0x437c820, 0x0}, {0xc00154a5e8, 0x16}, {0x4b4fad8, 0x18}, {0xc002ad2150, 0x6a}, 0x1176592e000)
	/go/src/github.com/modular-magician/terraform-provider-google-beta/google-beta/services/compute/compute_operation.go:107 +0x75
github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute.changeInstanceStatusOnCreation(0x4b03c56?, 0x7?, {0xc00154a5e8, 0x16}, {0xc002b7ffc0, 0xd}, {0x0, 0x0}, {0xc002ad2150, 0x6a})
	/go/src/github.com/modular-magician/terraform-provider-google-beta/google-beta/services/compute/resource_compute_instance.go:1454 +0x4a5
github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute.resourceComputeInstanceCreate(0x0?, {0x4ace060?, 0xc002976400})
	/go/src/github.com/modular-magician/terraform-provider-google-beta/google-beta/services/compute/resource_compute_instance.go:1558 +0x826
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x5548d20?, {0x5548d20?, 0xc001581230?}, 0xd?, {0x4ace060?, 0xc002976400?})
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:766 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000fda7e0, {0x5548d20, 0xc001581230}, 0xc0018ccd00, 0xc002521d80, {0x4ace060, 0xc002976400})
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:909 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc002618408, {0x5548d20?, 0xc0015811a0?}, 0xc001462410)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:1078 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x5548d58?, {0x5548d20?, 0xc001580f00?}, 0xc001462410)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002612820, {0x5548d20?, 0xc001580510?}, 0xc001f35570)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x4a66320?, 0xc002612820}, {0x5548d20, 0xc001580510}, 0xc002521200, 0x0)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000264e00, {0x5548d20, 0xc001580480}, {0x5554fe8, 0xc001cae480}, 0xc00149aea0, 0xc0029112f0, 0x78952d8, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1379 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000264e00, {0x5554fe8, 0xc001cae480}, 0xc00149aea0)
	/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1790 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 3225
	/go/pkg/mod/google.golang.org/grpc@v1.65.0/server.go:1040 +0x135
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute	10.351s
```

Reply:
Reinforced the condition because the function shouldn't be called in this scenario.